.github/workflows/test.yml

```diff
@@ -0,0 +1,21 @@
+name: Test
+
+run-name: Running tests and checks for plugin
+
+on:
+  push:
+  workflow_call:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+
```

Comment:
Suggestion to do linting here or possibly another job (though I kind of think here is fine personally, even if theory says get a whole new build agent and etc).